api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -2008,8 +2008,118 @@ type AzurePlatformSpec struct {
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// tenantID is the tenant ID of the Azure Key Vault on the management cluster.
+	//
+	// +kubebuilder:validation:Required
+	TenantID string `json:"tenantID"`
+}
+
+// AzureResourceManagedIdentities contains the managed identities needed for HCP control plane and data plane components
+// that authenticate with Azure's API.
+type AzureResourceManagedIdentities struct {
+	// controlPlane contains the client IDs of all the managed identities on the HCP control plane needing to
+	// authenticate with Azure's API.
+	//
+	// +kubebuilder:validation:Required
+	ControlPlane ControlPlaneManagedIdentities `json:"controlPlane"`
+
+	// Future placeholder - DataPlaneMIs * DataPlaneManagedIdentities
+}
+
+// ManagedIdentity contains the client ID, and its certificate name, of a managed identity. This managed identity is
+// used, by an HCP component, to authenticate with the Azure API.
+type ManagedIdentity struct {
+	// clientID is the client ID of a managed identity.
+	//
+	// +kubebuilder:validation:XValidation:rule="self.matches('^[{]?[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}[}]?$')",message="the client ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen separated hexadecimal characters in the form 8-4-4-4-12."
+	// +kubebuilder:validation:Required
+	ClientID string `json:"clientID"`
+
+	// certificateName is the name of the certificate backing the managed identity. This certificate is expected to
+	// reside in an Azure Key Vault on the management cluster.
+	//
+	// +kubebuilder:validation:Required
+	CertificateName string `json:"certificateName"`
+}
+
+// ControlPlaneManagedIdentities contains the managed identities on the HCP control plane needing to authenticate with
+// Azure's API.
+type ControlPlaneManagedIdentities struct {
+	// managedIdentitiesKeyVault contains information on the management cluster's managed identities Azure Key Vault.
+	// This Key Vault is where the managed identities certificates are stored. These certificates are pulled out of the
+	// Key Vault by the Secrets Store CSI driver and mounted into a volume on control plane pods requiring
+	// authentication with Azure API.
+	//
+	// More information on how the Secrets Store CSI driver works to do this can be found here:
+	// https://learn.microsoft.com/en-us/azure/aks/csi-secrets-store-driver.
+	//
+	// +kubebuilder:validation:Required
+	ManagedIdentitiesKeyVault ManagedAzureKeyVault `json:"managedIdentitiesKeyVault"`
+
+	// cloudProvider is a pre-existing managed identity associated with the azure cloud provider, aka cloud controller
+	// manager.
+	//
+	// +kubebuilder:validation:Required
+	CloudProvider ManagedIdentity `json:"cloudProvider"`
+
+	// clusterAPIAzure is a pre-existing managed identity associated with cluster-api azure.
+	//
+	// +kubebuilder:validation:Required
+	ClusterAPIAzure ManagedIdentity `json:"clusterAPIAzure"`
```

Comment:
Azure is redundant here. Also we might want to consolidate with the term we use in aws i.e NodePoolManagement, thoughts?

Reply:
> Also we might want to consolidate with the term we use in aws i.e NodePoolManagement, thoughts?
Although it is true that changing it to `NodePoolManagement` would consolidate with how it is in AWS, that name is not really accurate. The operator is the cluster api operator and not "node pool management".

Regarding "azure" being redundant, is it? my understanding is that we have capi (https://github.com/kubernetes-sigs/cluster-api-operator) , but also capz (https://github.com/kubernetes-sigs/cluster-api-provider-azure), and capa (https://github.com/kubernetes-sigs/cluster-api-provider-aws). Which one of them is representing?

Reply:
I like being explicit here since that MI is specifically for CAPZ and to avoid any confusion with CAPI or CAPA like Miguel alluded to. It's not a hill I'm willing to die on though 😄

Reply:
CAPI/CAPZ is an implementation detail we could technically transparently swap if we choose to. Therefore I'd prefer if we avoin baking a reference to it in the API, hence my suggestion to align with the aws naming we chose.
In any case I'm also ok with your preference if you feel strongly about it.

Reply:
I've updated this.